vip-application/src/main/java/fr/insalyon/creatis/vip/application/client/bean/boutiquesTools/BoutiquesApplication.java

```diff
@@ -0,0 +1,264 @@
+package fr.insalyon.creatis.vip.application.client.bean.boutiquesTools;
+
+import com.google.gwt.user.client.rpc.IsSerializable;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Representation of an application Boutiques descriptor
+ *
+ * @author Guillaume Vanel
+ * @version %I%, %G%
+ */
+public class BoutiquesApplication implements IsSerializable {
+
+    private String name;
+    private String description;
+    private String version;
+    private List<BoutiquesInput> inputs = new ArrayList<>();
+    // Input dependencies
+    private List<BoutiquesGroup> groups = new ArrayList<>();
+    private Map<String, List<String>> disablesInputsMap= new HashMap<>();
+    private Map<String, List<String>> requiresInputsMap= new HashMap<>();
+    private Map<String, Map<String, List<String>>> valueDisablesInputsMap= new HashMap<>();
+    private Map<String, Map<String, List<String>>> valueRequiresInputsMap= new HashMap<>();
+    // Other properties not used for launch form generation
+    private String applicationLFN;
+    private String author;
+    private String commandLine;
+    private String containerType;
+    private String containerImage;
+    private String containerIndex;
+    private String schemaVersion;
+    private String challengerEmail;
+    private String challengerTeam;
+    private List<BoutiquesOutputFile> outputFiles = new ArrayList<>();
+    private Map<String,String> tags = new HashMap<>();
+    private String jsonFile;
+
+    private BoutiquesApplication(){}
+
+    /**
+     * @param name String
+     * @param description String
+     * @param version String
+     */
+    public BoutiquesApplication(String name, String description, String version){
+        this.name = name;
+        this.description = description;
+        this.version = version;
+    }
+
+    /**
+     * @return String of format 'applicationName applicationVersion'
+     */
+    public String getFullName(){
+        return this.name + " " + this.version;
+    }
+
+    /**
+     * @return Application description as String
+     */
+    public String getDescription(){
+        return this.description;
+    }
+
+    /**
+     * @return Array of BoutiquesInputs representing application inputs
+     */
+    public List<BoutiquesInput> getInputs() {
+        return this.inputs;
+    }
+
+    /**
+     * @return Array of BoutiquesGroups representing application input groups
+     */
+    public List<BoutiquesGroup> getGroups() {
+        return groups;
+    }
+
+    /**
+     * @return Map representing input dependencies of type 'disables-inputs'. Keys are disabling input IDs as Strings,
+     * and values are arrays of Strings representing IDs of inputs disabled when disabling input is non empty
+     */
+    public Map<String, List<String>> getDisablesInputsMap() {
+        return this.disablesInputsMap;
```

Comment:
I do not like the idea to duplicate the maps from all boutiques inputs here (it is not exactly duplicate as they point to the same structures but still). 

I think all these getters on those map are only called once and so theses map fields can be replaced by inline construct like :
`return inputs.stream().collect(Collectors.toMap(BoutiquesInput::getId, BoutiquesInput::getDisablesInputsId));`